lib/usb_connection.js

```diff
@@ -70,17 +102,74 @@ USBConnection.prototype.open = function(callback) {
       if (error) return reject(error, self);
       self.epIn = self.intf.endpoints[0];
       self.epOut = self.intf.endpoints[1];
+      if (!self.epIn || !self.epOut) {
+        return reject(new Error("Device endpoints weren't able to be loaded"), self);
+      }
+
+      // Map desciptions
+      self.device.getStringDescriptor(self.device.deviceDescriptor.iSerialNumber, function (error, data) {
+        if (error) return reject(error, self);
+        self.serialNumber = data;
+
+        // Register this connection with daemon (keeps track of active remote processes)
+        Daemon.register(self);
+        // If all is well, resolve the promise with the valid connection
+        return resolve(self);
+      })
     });
-    // Map desciptions
-    self.device.getStringDescriptor(self.device.deviceDescriptor.iSerialNumber, function (error, data) {
-      if (error) return reject(error, self);
-      self.serialNumber = data;
-      // If all is well, resolve the promise with the valid connection
-      return resolve(self);
-    })
   });
 }
 
+USBConnection.prototype.end = function(callback) {
+  var self = this;
+  // Tell the USB daemon to end all processes active
+  // on account of this connection
+  Daemon.deregister(this, function(err) {
+    self._close();
+    callback && callback(err);
+  });
+}
+
+USBConnection.prototype._close = function(callback) {
+  var self = this;
+
+  if (self.closed) {
+    return callback && callback();
+  }
+
+  self.closed = true;
+  self.intf.release(true, function (err) {
+    self.device.close();
+    callback && callback();
+  }.bind(self));
+}
+
+USBConnection.prototype._processArgsForTransport = function(command) {
+
+  if (!Array.isArray(command)) return;
+
+  function quoteRemover(match, p1, p2, p3) {
+    return p2;
+  }
+
+  // For each command
+  command.forEach(function(arg, i) {
```

Comment:
Things passed to this shouldn't be quoted. Escape strings for SSH transport, don't unescape them here. It only needs to check that the array items don't themselves contain null bytes, but can probably just reject those.